frontend/src/contexts/authenticatedContext.jsx

```diff
@@ -1,17 +1,44 @@
-import React, { createContext, useState, useContext } from "react";
+import React, { createContext, useState, useContext, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
 
-const authenticatedContext = createContext([]);
-export const useAuthenticated = () => useContext(authenticatedContext);
+const authenticatedContext = createContext();
+export const useAuthenticatedContext = () => useContext(authenticatedContext);
+const BACKEND_URL = import.meta.env.VITE_BACKEND_URL;
 
-const AuthenticatedContextProvider = ({ children }) => {
+export const AuthenticatedContextProvider = ({ children }) => {
   const [isUserAuthenticated, setIsUserAuthenticated] = useState(false);
+  const navigate = useNavigate();
+  useEffect(() => {
+    const checkAuth = async () => {
```

Comment:
Similar feedback as previous PR to not nest functions inside other functions

Reply:
Ok, in the future best practice is to make changes from feedback in the same commit, not a separate commit